src/auto-clickers/auto-clicker.js

```diff
@@ -1,42 +1,128 @@
     name = "";
+    examineText = "";
+    cost = 0;
     dps = 0;
     level = 0;
 
-    scene;
+    // Scenes
+    scrollWindow;
+    chat;
+    dashboard;
+    currentScene;
+    stats;
 
     damageInterval = 0;
     timer = {};
+    text = {};
 
     constructor(data) {
-        console.log("Creating " + data.name + " autoclicker with " + data.dps + " dps");
-        // Pull into local objects
-        this.scene = data.scene;
-        this.stats = data.scene.stats;
+        this.scrollWindow = data.scene;
+        this.examineText = data.examineText;
+        this.cost = data.cost;
         this.dps = data.dps;
         this.level = data.level;
-        this.name = name.type;
+        this.name = data.name;
 
-        // Damage every .1 second
-        this.damageInterval = 0.1;
-        this.timer = setInterval(this.clickTarget.bind(this), this.damageInterval * 1000);
+        // Damage every .5 seconds
+        this.damageInterval = 100;
+        this.timer = this.scrollWindow.time.addEvent({
+            delay: this.damageInterval,
+            callback: () => {
+                this.clickTarget();
+            },
+            loop: true,
+            paused: true,
+        });
+    }
+
+    start(currentScene) {
+        this.timer.paused = false;
+        this.currentScene = currentScene;
+        this.stats = currentScene.stats;
+
+        this.stats.updateAutoClickerDPS(this.dps);
     }
 
     clickTarget() {
-        let damagePerTick = this.dps * this.damageInterval;
-        this.scene.clickCurrentTarget(damagePerTick);
+        let damagePerTick = this.dps * (this.damageInterval / 1000);
+        this.currentScene.clickCurrentTarget(damagePerTick);
         this.stats.updateAutoClickDamageStat(damagePerTick);
     }
 
-    release() {
-        clearInterval(this.timer);
+    createText(isShop = false, x = 0, y = 0) {
+        this.text = this.scrollWindow.add
+            .text(x, y, this.name, { font: "16px runescape" })
+            .setDepth(4)
+            .setInteractive()
+            .setOrigin(0, 0)
+            .on("pointerover", () => {
+                this.examine(isShop);
+            })
+            .on("pointerout", () => {
+                if (this.chat != undefined) {
+                    this.chat.showObjectInfo(false);
+                }
+            })
+            .on("pointerup", () => {
+                if (isShop && this.scrollWindow.characterData.gold >= this.cost) {
+                    this.buy();
+                }
+            });
+        this.displayHeight = this.text.displayHeight;
+    }
+
+    async buy() {
+        this.scrollWindow.characterData.gold -= this.cost;
+        if (this.dashboard == undefined) {
+            this.dashboard = this.scrollWindow.scene.get(CONSTANTS.SCENES.DASHBOARD);
+        }
+        let newMember = await getAutoclickerClass(this.name, this.dashboard);
+        this.dashboard.clan.obj.addClanMember(newMember);
+    }
+
+    examine(isShop) {
```

Comment:
This goes for all examines - not just this one - but the right click menu should close when examine is clicked.

Reply:
How would you close the examine window then? Currently I have it so closing the right click menu closes the examine window as well

Reply:
If we change examines to be a log to chat then it would be fine

Reply:
Yeah, do you have any good ideas on where the close/open chat button would be? I think the chat log would open automatically when something is appended to it. But you should be able to open-close it at will other times

Reply:
Addressed here:
https://github.com/CurtisGreen/rs-clicker/pull/6